src/refdb_fs.c

```diff
@@ -63,6 +66,18 @@ typedef struct refdb_fs_backend {
 	uint32_t direach_flags;
 } refdb_fs_backend;
 
+/*
+ * List of symbolic reference names that are always stored in the
+ * gitdir. This is required such that we are able to correctly
+ * resolve these references via the gitdir instead of the
+ * commondir when working trees are used.
+ */
+static const char *symrefs[] = {
+    "HEAD",
+    "ORIG_HEAD",
+    NULL,
+};
```

Comment:
They are pseudorefs, as @carlosmn previously pointed out (see [gmane](http://permalink.gmane.org/gmane.comp.version-control.git/275036)).
The list was not complete yet, wanted to come back to it later. Should've left a TODO in there, I guess. I don't exactly know how pseudorefs are handled yet, I'll have to figure that out.